pkg/asset/manifests/openshift.go

```diff
@@ -154,9 +155,16 @@ func (o *Openshift) Generate(dependencies asset.Parents) error {
 			},
 		}
 	case vspheretypes.Name:
+		vCenterURL, err := url.Parse(installConfig.Config.VSphere.VCenter)
+		host := installConfig.Config.VSphere.VCenter
+		if err == nil {
+			if hostname := vCenterURL.Hostname(); hostname != "" {
+				host = hostname
+			}
+		}
```

Comment:
this also is not working for the same reasons I pointed out in the other comment: https://github.com/openshift/installer/pull/3836#discussion_r451723695

you can test this like this:

1. create an install config which includes a port 
```console
$ cat install-config.yaml | grep vCenter
    vCenter: vcsa.vmware.devcluster.openshift.com:6443
```
2 create manifests
```
$ openshift-install create manifests
```

3 check the manifest

```
$ cat openshift/99_cloud-creds-secret.yaml 
kind: Secret
apiVersion: v1
metadata:
  namespace: kube-system
  name: vsphere-creds
data:
  vcsa.vmware.devcluster.openshift.com:6443.username: <redacted>
  vcsa.vmware.devcluster.openshift.com:6443.password: <redacted>

```

the goal here is that `vcsa.vmware.devcluster.openshift.com:6443.username` would be `vcsa.vmware.devcluster.openshift.com.username`